go-app-ussd_higherhealth_healthcheck.js

```diff
@@ -1701,32 +1702,81 @@ go.app = (function () {
         return self.states.create(name, opts);
       });
     });
-
-    self.states.add("state_welcome", function(name) {
+    self.states.add("state_welcome", function(name, opts) {
       self.im.user.answers.google_session_token = crypto.randomBytes(20).toString("hex");
       var question;
-      if (self.im.user.answers.returning_user) {
+      var date = moment().format('YYYY-MM-DD');
+      var msisdn = self.im.user.addr;
+      return new JsonApi(self.im).get(
+        // use dictionary params
+        self.im.config.eventstore.url + "/api/v3/covid19triage/", {
+        headers: {
+          "Authorization": ["Token " + self.im.config.eventstore.token],
+          "User-Agent": ["Jsbox/HH-Covid19-Triage-USSD"]
+        },
+        params: {
+          "timestamp_gt": date + "T00:00:00+02:00",
+          "msisdn": msisdn
+        }
+      }).then(function (response) {
+        var results = response.data.results;
+        if(results.length === 0){
+          if (self.im.user.answers.returning_user){
+            question = $([
+              "Welcome back to HIGHER HEALTH's HealthCheck.",
+              "No result SMS will be sent. Continue or WhatsApp HI to 0600110000",
+              "",
+              "Reply"
+            ].join("\n"));
+          }
+          else {
+            question = $([
+              "HIGHER HEALTH HealthCheck is your risk assessment tool.",
+              "No result SMS will be sent. Continue or WhatsApp HI to 0600110000",
+              "",
+              "Reply"
+            ].join("\n"));
+          }
+          return new MenuState(name, {
+            question: question,
+            error: $("This service works best when you select numbers from the list"),
+            accept_labels: true,
+            choices: [
+              new Choice("state_terms", $("START"))
+            ]
+          });
+        }
+        var last_result = results[results.length - 1];
+        self.im.user.answers.risk = last_result.risk;
+        self.im.user.answers.state_first_name = last_result.first_name;
+        self.im.user.answers.state_last_name = last_result.last_name;
```

Comment:
Looking at `state_display_risk`, it seems we also need the `state_tracing` answer